### ISSUE TITLE
[🐞] code used only by a `loader$` gets loaded in the browser

### ISSUE THREAD

**Issue by DustinJSilk** [BuilderIO/qwik#2733]
### Which component is affected?

Qwik City (routing)

### Describe the bug

If a file exports 2 functions, and one is used only by a `loader$` and the other is used only by a component, during SPA navigation, the entire file is loaded into the browser.

This will cause an error if the file also uses server only code or imports a server package.
It also could potentially leak sensitive code.

The issue occurs during development with `pnpm start`, Im also experiencing it on my preview server.

**This is using the latest code from qwik-build and qwik-city-build**

### Reproduction

See below

### Steps to reproduce

Create a file that has 2 pieces of logic:
```ts
// Some backend code
export const authenticateUser = async (): Promise<string> => {
  console.log("This should not run in the browser");

  return "my-data";
};

// A hook that can run on both environments
export function useAuth(userData: string) {
  const user = useSignal<{ user: string }>({ user: userData });

  useContextProvider(USER_CONTEXT, user.value);
}
```

Then in a child route, use both pieces of logic:

```ts
export const childLoader = loader$(authenticateUser);

export default component$(() => {
  const { value } = childLoader.use();
  useAuth(value);

  return ...
});
```

Finally, in your browser, navigate to the child page using a Link and you will be able to search for "This should not run in the browser" in your browser.

### System Info

```shell
System:
    OS: macOS 12.0.1
    CPU: (12) x64 Intel(R) Core(TM) i7-9750H CPU @ 2.60GHz
    Memory: 23.43 MB / 16.00 GB
    Shell: 5.8 - /bin/zsh
  Binaries:
    Node: 18.12.1 - ~/.nvm/versions/node/v18.12.1/bin/node
    Yarn: 1.22.18 - /usr/local/bin/yarn
    npm: 7.13.0 - /usr/local/bin/npm
  Browsers:
    Chrome: 109.0.5414.119
    Firefox: 109.0
    Safari: 15.1
  npmPackages:
    @builder.io/qwik: BuilderIO/qwik-build#main => 0.16.2 
    @builder.io/qwik-city: BuilderIO/qwik-city-build#main => 0.1.0-beta9 
    undici: 5.16.0 => 5.16.0 
    vite: 4.0.4 => 4.0.4
```

**Comment by manucorporat** [comment]
can you prepare a small repo case? during prod build it should be treeshaken!

The issue during `npm start` is expected, we don't do treeshaking during development.

Inlining the logic in `loader$()` will prevent many of this issues:

```tsx
export const childLoader = loader$(async (): Promise<string> => {
  console.log("This should not run in the browser");

  return "my-data";
});
```

**Comment by DustinJSilk** [comment]
Ok try this with a preview build! 
You should see one chunk is like 1.5mbs and when you navigate twice with the buttons theres an error in the console which will point you to a loader$s package being imported.

https://github.com/DustinJSilk/qwik-loader-treeshaking

I've inlined my loaders now so its not a high priority for me particularly

**Comment by manucorporat** [comment]
I cant reproduce the issue in the repo you provided, the code does not leak to the browser in preview mode

**Comment by DustinJSilk** [comment]
Thats odd.. Would you mind taking a look at this recording where the issue exists?
https://drive.google.com/file/d/1Y0JQGeiWV0WWCUaexrJ3QgpbzfgyNiRI/view

**Comment by wmalarski** [comment]
Hello. I worked with those kinds of issues before. I think changing `loader$` to this should help:
```ts
export const childLoader = loader$(() => {
  authenticateUser();
});
```
After this change `auth.ts` is not fetched on the client.

I assume this is related to the compiler analysis of AST. By making this arrow function -> splitting server/client code from server-only is easier. I hope this is the correct way of thinking about this.

**Comment by bayssmekanique** [comment]
I'm experiencing this same issue in my code base as well. I have had to wrap some of my server code in `server$` to prevent it from leaking to the client.  I tried to reproduce on [StackBlitz](https://stackblitz.com/edit/qwik-starter-epzklk?file=src/routes/index.tsx) but was unsuccessful.

I tested using a `routerAction$`, `routeLoader$`, and `useVisibleTask$` all in the same component which I suspect is the issue in my code base, but it works fine on StackBlitz.  If I eliminate `useVisibleTask$` in my component the issue goes away, so it definitely seems to be a relationship between the loader and the `useVisibleTask$` call.
